### PR TITLE
feat(medusa): Remove unnecessary count queries when not needed

### PR DESCRIPTION
**Pull request by adrien2p**
FIXES CORE-1397

### CODE REVIEW

File: packages/medusa/src/repositories/money-amount.ts [hunk 1]
```diff
@@ -328,10 +338,22 @@ export const MoneyAmountRepository = dataSource
...
       }
 
-      const [prices, count] = await qb.getManyAndCount()
+      let prices
+      let count = 0
+
+      if (shouldCount) {
+        const results = await qb.getManyAndCount()
+        prices = results[0]
+        count = results[1]
```
Comment (olivermrbl) [inline packages/medusa/src/repositories/money-amount.ts:347]: todo: omit count

**Review by olivermrbl — CHANGES_REQUESTED** [review]

File: packages/medusa/src/repositories/money-amount.ts (hunk 1 above)
Comment (adrien2p) [reply packages/medusa/src/repositories/money-amount.ts:347]: If shouldCount is true we should assign the count. I am not sure to follow

Comment (olivermrbl) [reply packages/medusa/src/repositories/money-amount.ts:347]: haha sorry, this was an old comment - ignore 🤦

**Comment by adrien2p** [comment]
Keep for later

**Pull request closed without merging**